datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -1082,34 +1219,38 @@ private void waitForTaskRelease(DatastreamTask task, long timeoutMs, String lock
    * @see #releaseTask(DatastreamTaskImpl)
    */
   public void acquireTask(DatastreamTaskImpl task, Duration timeout) {
-    _zkclient.ensurePath(KeyBuilder.datastreamTaskLockRoot(_cluster, task.getConnectorType()));
-    String lockPath = KeyBuilder.datastreamTaskLock(_cluster, task.getConnectorType(), task.getDatastreamTaskName());
+    _zkclient.ensurePath(KeyBuilder.datastreamTaskLockPrefix(_cluster, task.getConnectorType(), task.getTaskPrefix()));
+    String lockPath = KeyBuilder.datastreamTaskLock(_cluster, task.getConnectorType(), task.getTaskPrefix(), task.getDatastreamTaskName());
     String owner = null;
     if (_zkclient.exists(lockPath)) {
-      owner = _zkclient.ensureReadData(lockPath);
-      if (owner.equals(_instanceName)) {
+      owner = _zkclient.readData(lockPath, true);
```

Comment:
The reason to change from readData to ensureReadData is earlier it made sense, since the node was ephemeral and the node creation and deletion was happening in one instance code path. This will not be beneficial in case of persistent and multiple instances can try to delete (self-delete, or anyone trying to acquire lock after debounce timer.) I thought the api name was self-explaintary with the parameter.

Reply:
Thanks for explaining offline, make sense to me now. Since leaving it as ensureReadData() causes tests to fail I'm okay with either adding a comment or leaving it out to explain why readData() is used here.